src/core/include/mp-units/math.h

```diff
@@ -198,6 +198,24 @@ template<auto R, auto S, auto T, typename Rep1, typename Rep2, typename Rep3>
 }
 
 
+/**
+ * @brief Computes the floating-point remainder of the division operation x / y.
+ */
+template<auto R1, typename Rep1, auto R2, typename Rep2>
+[[nodiscard]] constexpr QuantityOf<get_quantity_spec(R1)> auto fmod(const quantity<R1, Rep1>& x,
+                                                                    const quantity<R2, Rep2>& y) noexcept
+  requires requires(Rep1 v1, Rep2 v2) {
+    common_reference(R1, R2);
+    requires requires { fmod(v1, v2); } || requires { std::fmod(v1, v2); };
+  }
+{
```

Comment:
Please put the constraints before a function signature to be consistent with the rest of the files. We use trailing requires clauses only for non-template member functions or for cases where we use function argument identifiers.

```suggestion
template<auto R1, typename Rep1, auto R2, typename Rep2>
  requires requires(Rep1 v1, Rep2 v2) {
    common_reference(R1, R2);
    requires requires { fmod(v1, v2); } || requires { std::fmod(v1, v2); };
  }
[[nodiscard]] constexpr QuantityOf<get_quantity_spec(R1)> auto fmod(const quantity<R1, Rep1>& x,
                                                                    const quantity<R2, Rep2>& y) noexcept
{
```

Reply:
Agh, fixed this on the `atan2`'s but these big function declarations just look like a wall of text to me. Can't wait for contracts to make it even better...

Reply:
Contracts will not make it better. On the contrary, the function signatures will be even longer 😉 
But I hope that at some point, we will get CPOs for math functions, as I discussed this with @mattkretz several times already :wink:

Reply:
Full sarcasm there, contracts will make it a nightmare :joy: Maybe I have spent too much time with Java but annotations always read a lot easier than anything we have in C++, and the latest concepts proposal seems particularly awful to read.

CPOs for math sounds great, I guess they are no different to overloaded operators.

Reply:
With CPOs we will not have to check for alternatives in the requires clause because ADL is an implementation detail of CPO rather than a user-facing interface requirement.